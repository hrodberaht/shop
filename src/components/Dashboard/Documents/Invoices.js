import React, { Component } from 'react';
import AddInvoiceForm from './AddInvoiceForm';

export default class Invoices extends Component {
  state = {
    isVisableAddInvoice: false,
  };

  static propTypes = {};

  handleClick = () => {
    this.setState(state => ({ isVisableAddInvoice: !state.isVisableAddInvoice }));
  };

  render() {
    return (
      <div>
        <button type="button" onClick={this.handleClick}>
          Add invoice
        </button>
        {this.state.isVisableAddInvoice && <AddInvoiceForm />}
      </div>
    );
  }
}
