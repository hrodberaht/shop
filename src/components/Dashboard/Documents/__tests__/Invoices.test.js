import Invoices from '../Invoices';

const defaultProps = {};

const setup = buildSetup(Invoices, defaultProps);

describe('<Invoices />', () => {
  it('should render without crash', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });

  it('initial state isVisibleAddInvoice should be false ', () => {
    const { wrapper } = setup();
    expect(wrapper.state().isVisibleAddInvoice).toBe(false);
  });

  it('should show form if state isVisibleAddInvoice is true', () => {
    const { wrapper } = setup();
    wrapper.setState({ isVisibleAddInvoice: true });
    expect(wrapper).toMatchSnapshot();
  });

  it('state isVisibleAddInvoice should be true after handleClick call ', () => {
    const { wrapper } = setup();
    wrapper.instance().handleClick();
    expect(wrapper.state().isVisibleAddInvoice).toBe(true);
  });
});
