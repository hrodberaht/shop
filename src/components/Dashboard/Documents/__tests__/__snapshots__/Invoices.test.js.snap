// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Invoices /> should render without crash 1`] = `
<div>
  <button
    onClick={[Function]}
    type="button"
  >
    Add invoice
  </button>
</div>
`;

exports[`<Invoices /> should show form if state isVisableAddInvoice is true 1`] = `
<div>
  <button
    onClick={[Function]}
    type="button"
  >
    Add invoice
  </button>
  <Connect(ReduxForm) />
</div>
`;
